TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/DriveSystem.java

```diff
@@ -162,26 +168,60 @@ public boolean driveToPositionTicks(int ticks, Direction direction, double maxPo
         }
 
         for (DcMotor motor : motors.values()) {
+//            Log.d(TAG, motor.toString() + ", " + motor.getPortNumber() + ": " + motor.getCurrentPosition());
             int offset = Math.abs(motor.getCurrentPosition() - mTargetTicks);
-            if(offset <= 0){
-
+            if(offset <= 15){
                 // Shut down motors
                 setMotorPower(0);
-
                 // Reset motors to default run mode
                 setRunMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
-
                 // Reset target
                 mTargetTicks = 0;
-
                 // Motor has reached target
                 return true;
             }
         }
 
+        double currHeading = imuSystem.getHeading();
+        Log.d("Curr heading: ", currHeading + " ");
+        if (strafeSet && mStrafeHeading != currHeading && Direction.isStrafe(direction)) {
+            double diff = computeDegreesDiff(mStrafeHeading, currHeading);
```

Comment:
Aren't `computeDegreesDiff` and `getError` essentially the same? And do either of them actually need parameters?

Reply:
I changed it a bit - is the way it is now better? Also, parameter justification: if we pass in the two angles as parameters, it allows for broader use and possibilities of being used in different ways in client programs like BaseStateMachine.

Reply:
I think this should be a private helper method within the class. And both parameters are available within the method. (And there should only be one helper method.) (And I don't think we need to worry about being 180 degrees off.)